### DISCUSSION TITLE
Ability to upload Cost and Usage metrics in Langfuse for Gemini Models from LlamaIndex

### DISCUSSION THREAD

**Discussion by ansu86d** [langfuse/langfuse#2736] [Support]
Hi Team,

I have a sample app using LlamaIndex and the ReAct framework(for reasoning and action) , using Function Calling for text to SQL functionality 
I am able to get the traces , but there are 2 observations 

1. Unable to view the cost and usage as I am using Gemini model, , kindly share the code which I could use along with LlamaIndex , I believe there are some API's but unable to get it working 
2. Getting an error post run of the below code 
Error:
AttributedList.__init__() takes 1 positional argument but 2 were given
Traceback (most recent call last):
  File "\Python311\Lib\site-packages\langfuse\client.py", line 2122, in span
    "body": event.dict(exclude_none=True),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\Python311\Lib\site-packages\langfuse\api\resources\ingestion\types\create_span_body.py", line 30, in dict
    return super().dict(**kwargs_with_defaults)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\Python311\Lib\site-packages\langfuse\api\resources\ingestion\types\create_event_body.py", line 28, in dict
    return super().dict(**kwargs_with_defaults)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\Python311\Lib\site-packages\langfuse\api\resources\ingestion\types\optional_observation_body.py", line 43, in dict
    return super().dict(**kwargs_with_defaults)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\Python311\Lib\site-packages\pydantic\v1\main.py", line 449, in dict
    return dict(
           ^^^^^
  File "\Python311\Lib\site-packages\pydantic\v1\main.py", line 866, in _iter
    v = self._get_value(
        ^^^^^^^^^^^^^^^^
  File "\Python311\Lib\site-packages\pydantic\v1\main.py", line 760, in _get_value
    return {
           ^
  File "\Python311\Lib\site-packages\pydantic\v1\main.py", line 761, in <dictcomp>
    k_: cls._get_value(
...
  File "\Python311\Lib\site-packages\pydantic\v1\main.py", line 793, in _get_value
    return v.__class__(*seq_args) if is_namedtuple(v.__class__) else v.__class__(seq_args)
                                                                     ^^^^^^^^^^^^^^^^^^^^^
TypeError: AttributedList.__init__() takes 1 positional argument but 2 were given

---------------------------------------------------------------------------------------------------

Code : 

from llama_index.core import Settings
from llama_index.core.callbacks import CallbackManager
from langfuse.llama_index import LlamaIndexCallbackHandler
import json
from llama_index.llms.openai import OpenAI
from google.oauth2 import service_account
from llama_index.llms.vertex import Vertex
from llama_index.core.agent import ReActAgent

langfuse_callback_handler = LlamaIndexCallbackHandler(
    public_key="pk-XXX",
    secret_key="sk-XXX",
    host="https://cloud.langfuse.com",
    user_id="XXX",
    session_id="XXX",
    tags=["XXX"]
)

Settings.callback_manager = CallbackManager([langfuse_callback_handler])

filename = "XXX.json"

credentials: service_account.Credentials = (
    service_account.Credentials.from_service_account_file(filename)
)

llm = Vertex(model="gemini-1.5-pro-001", project=credentials.project_id, credentials=credentials) 
Settings.llm = llm

def fn_db_tools():
    db_tools = DatabaseToolSpec(
        scheme="postgresql",  # Database Scheme
        host="localhost",  # Database Host
        port="5432",  # Database Port
        user="XXXX",  # Database User
        password="XXXX",  # Database Password
        dbname="XXXX",  # Database Name
    )
    return db_tools

def fn_call_react_agent(input):
    agent = ReActAgent.from_tools(fn_db_tools().to_tool_list(), llm=llm, verbose=False)
    return agent.chat(input).response

input = "List of customers name starting with A ?"
output = fn_call_react_agent(input)
print(output)

---------------------------------------------

langfuse :  2.39.3
llamaindex: 0.10.50

**Comment by ansu86d** [comment]
@marcklingen : Adding you for reference based on the discussion

**Reply by marcklingen** [reply]
thanks for sharing! it is expected that the USD costs for gemini models are not working yet as we need to add gemini specific billing logic. The latter error looks odd though, are you on the latest Langfuse SDK version? Can you open a bug report for this problem?

**Reply by jahidhasanlinix** [reply]
Hi, I was using the Gemini model. Can long fuse support Gemini for model usage, cost tracking, as well as prompt and other evals? I am using LangGraph langchain_google_genai library
